kitsune/sumo/static/sumo/js/device-migration-wizard.js

```diff
@@ -3,6 +3,37 @@ import trackEvent from "sumo/js/analytics";
 import SwitchingDevicesWizardManager from "sumo/js/switching-devices-wizard-manager";
 import "sumo/js/form-wizard";
 
-$(document).ready(function() {
-  new SwitchingDevicesWizardManager(document.querySelector("#switching-devices-wizard"))
+document.addEventListener("DOMContentLoaded", function () {
+  new SwitchingDevicesWizardManager(document.querySelector("#switching-devices-wizard"));
+  kbTabsInit();
 });
+
+
+// This function initializes the tabs on the switching-devices page
+// and adds click event listener
+function kbTabsInit() {
+  document.getElementById("tab-view-all").style.display = "block";
+  let tabItems = document.querySelectorAll('.tabs--link');
+  tabItems.forEach((item) => {
+    item.addEventListener("click", (ev) => {
+      toggleTabContent(ev);
+    });
+  });
+}
+
+// This function toggles the visibility of the tab content based on 
+// clicked tab 
+function toggleTabContent(ev) {
+  let tabContent = document.getElementsByClassName("topic-list");
+  [].forEach.call(tabContent, (el) => {
+    el.style.display = "none";
+  });
+
+  let tabItems = document.getElementsByClassName("tabs--link");
+  [].forEach.call(tabItems, (item) => {
+    item.classList.remove("is-active");
+  });
+
+  ev.target.parentElement.classList.add("is-active");
```

Comment:
I think if I had any critique, it'd be here. This is the SUMO team's codebase, and so I defer to your styles, preferences and idioms - but what we tend to do on the Firefox side of things for stuff like this is put most of the hiding and showing rules in our CSS, and then use classes / attributes from script. So what I've highlighted here, we might have as:

```css
.topic-list:not(.is-active) {
  display: none;
}
```

```js
// Remove the active state from the previously selected tab / tab content
let oldSelectedTabContent = document.querySelector(".topic-list.is-active");
let oldSelectedTab = document.querySelector(".tabs--link.is-active");

oldSelectedTabContent.classList.remove("is-active");
oldSelectedTab.classList.remove("is-active");

// Put the newly selected tab / tab content into the active state.
let selectedTab = ev.target.parentElement;
let selectedTabContent = document.querySelector(`#tab-${ev.target.parentElement.dataset.eventLabel}`);

selectedTab.classList.add("is-active");
selectedTabContent.classList.add("is-active");
```

(I haven't tested this code, but hopefully it conveys the general idea)

I think the reason why we tend to avoid setting styles inline with script is because it keeps CSS as the domain where we declare the style characteristics of things, which helps separate concerns a little bit. (Inline CSS from script is also more difficult to override from stylesheets without hacks like `!important`.)

This is a minor critique and, I think, ultimately a matter of preference and convention. I wouldn't block on my suggestion.

Reply:
I think that what @mikeconley says here makes sense; separating concerns is never a bad move.

Reply:
Ah, I think `getElementById(id)` handles spaces in the `id` just fine, but if I'm not mistaken I think `querySelector(cssSelector)` would not if the `cssSelector` contained spaces.

Reply:
So I guess my concern boils down to this:
- Using the `subtopic.title` within HTML element `id`s may introduce spaces into the `id`s
- Spaces in `id`s make them difficult/problematic to query -- you'd have to know to escape them if querying by CSS selector, or use `getElementById` instead

Reply:
Thanks @mikeconley that's great feedback. I restructured the HTML a bit to make room for the snippet above. It's much cleaner now.